Readability analysis

georgerichardson opened the pull request:
---

Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [ ] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [ ] I have explained the feature in this PR or (better) in `output/reports/`
- [x] I have requested a code review

closes #20 #28

georgerichardson committed:
fix kwarg for saving pickle

georgerichardson committed:
Add specific input and output data

georgerichardson committed:
Remove inputs

georgerichardson committed:
Rebase onto dev

georgerichardson committed:
Add readability table and figure outputs

Jack-0-0 commented on comp_sci_gender_bias/pipeline/readbility/subj_readability.py:
```diff
@@ -0,0 +1,212 @@
+from tokenize import Name
```
Unused import

Jack-0-0 commented on comp_sci_gender_bias/pipeline/readbility/subj_readability.py:
```diff
@@ -0,0 +1,212 @@
+from typing import List
+
+from comp_sci_gender_bias import PROJECT_DIR
+from comp_sci_gender_bias.getters.school_data import text_descriptions
```
Unused import

Jack-0-0 commented on comp_sci_gender_bias/pipeline/readbility/subj_readability.py:
```diff
@@ -0,0 +1,212 @@
+    Returns:
+        Dataframe of clean course descriptions for all subjects.
+    """
+    scraped = pd.read_csv(SCRAPED_DIR)
```
This could be replaced by using the getter at `getters.scraped_data.scraped_data`

Jack-0-0 commented on requirements.txt:
```diff
@@ -13,3 +13,9 @@ cyhunspell
 wget
 gensim
 sentence-transformers
+<<<<<<< HEAD
+python-dotenv
+=======
+textstat
+tabulate
+>>>>>>> 8986a7c (Add textstat and tabulate)
```
There are some conflict markers here

Jack-0-0 reviewed (approved):
Thanks @georgerichardson, the code looks good. The code runs and produces the same outputs that you've produced. I have added some comments on specific lines.

There's no readme relating to `pipeline/readbility/subj_readability.py`.

For `pipeline/readbility/subj_readability.py`, should some of this file be in pipeline and the chart making be in analysis?

There's a typo in the directory name readability at `pipeline/readbility/`

For the boxplots, what do the diamonds represent?

It looks like the PR needs to be rebased to dev.

Jack-0-0 commented on outputs/tables/readability/quantile_examples/quantile_examples.md:
```diff
@@ -0,0 +1,11 @@
+|    | Score                  | Subject   |   Quantile | Example                                                                                                                                                                                                                                                                                                                                                ...
+|---:|:-----------------------|:----------|-----------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ ...
```
What do these dashed lines represent?

georgerichardson replied:
In a markdown table, these represent the dividen between the header row and the table rows. The colons indicate which way to align the content of that row.

Jack-0-0 replied:
Thanks, I hadn't seen it before.

georgerichardson committed:
Add steps for readability pipeline

georgerichardson committed:
Fix dir typo

georgerichardson commented:
> Thanks @georgerichardson, the code looks good. The code runs and produces the same outputs that you've produced. I have added some comments on specific lines.

Thanks. Now addressed.

> There's no readme relating to pipeline/readbility/subj_readability.py.

Now updated

> For pipeline/readbility/subj_readability.py, should some of this file be in pipeline and the chart making be in analysis?

Probably a good idea, but given the time I think we should keep it like this for now. Happy to change if you want things to be proper though!

> There's a typo in the directory name readability at pipeline/readbility/

Good spot!

> For the boxplots, what do the diamonds represent?

Outliers ([see Seaborn docs](https://seaborn.pydata.org/generated/seaborn.boxplot.html))

> It looks like the PR needs to be rebased to dev.

Thanks. I hope I've fixed this. I think I used a dodgy commit hash last time.

Jack-0-0 commented on comp_sci_gender_bias/pipeline/README.md:
````diff
@@ -101,3 +101,13 @@ python comp_sci_gender_bias/pipeline/school_level_gender_bias/make_school_lvl_ge
+
+To calculate the readability scores and generate the chart and table outputs, run:
+
+```bash
+python comp_sci_gender_bias/pipeline/readbility/subj_readability.py
+```
````
```suggestion
```bash
python comp_sci_gender_bias/pipeline/readability/subj_readability.py
```
```

georgerichardson committed:
Fix typo in readability pipeline dir
